server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -163,6 +164,19 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a sequential unordered stream of the map entries.
+     *
+     * @return a {@link Stream} of the map entries as {@link Entry}
+     */
+    public Stream<Entry<KType, VType>> stream() {
+        Stream.Builder<Entry<KType, VType>> streamBuilder = Stream.builder();
+        for (ObjectObjectCursor<KType, VType> cursor : map) {
+            streamBuilder.add(new Entry<>(cursor.key, cursor.value));
```

Comment:
I think this essentially copies the map into a `SpinedBuffer` and I think that can be avoided if we instead create our own spliterator to back the stream? I think using `AbstractSpliterator` will work, needing just a `tryAdvance` that could do something like:

```
            public boolean tryAdvance(Consumer<? super E> action) {
                if (this.iterator.hasNext()) {
                    ObjectObjectCursor<K,V> cursor = this.iterator.next()
                    action.accept(new Map.Entry<>(cursor.key, cursor.value));
                    return true;
                } else {
                    return false;
                }
            }

```

Reply:
Awesome! I replaced `SpinedBuffer` with a sized `AbstractSpliterator`